learn/how-tos/Integrating-JUnit-tests-for-WaveMaker-application.md

```diff
@@ -0,0 +1,128 @@
+# Integrating JUnit tests for WaveMaker application
+
+
+### OVERVIEW
+
+JUnit is a unit testing framework for the Java programming language. JUnit has been important in the development of test-driven development. These tests are required whether imported services are working as expected or not, before proceeding to run/deployment of the application. This document demonstrates how to integrate JUnit tests in an application developed by WaveMaker RAD.
+
+### GOALS
+
+1. Integrate JUnit tests in the WaveMaker application
+2. Writing sample test cases for services.
+
+### PREREQUISITES 
+
+1. IDE (IntelliJ IDEA, Eclipse)
+2. WaveMaker Application
+3. Git
+4. Maven
+
+### CONFIGURATION
+
+1. Export the WaveMaker application as a zip file.
+2. Import this zip file into IDE (Use [WaveMaker workspace sync plugin](/learn/how-tos/synchronizing-wavemaker-apps-ides-beta)).
+3. Add the following dependencies in pom.xml file.
+
+```    
+    <dependency>
+        <groupId>org.springframework</groupId>
+        <artifactId>spring-test</artifactId>
+        <scope>provided</scope>
+    </dependency>
+    <dependency>
+        <groupId>junit</groupId>
+        <artifactId>junit</artifactId>
+        <version>4.13.2</version>
+        <scope>provided</scope>
+    </dependency>
+    <dependency>
+        <groupId>org.hamcrest</groupId>
+        <artifactId>hamcrest-all</artifactId>
+        <version>1.3</version>
+        <scope>provided</scope>
+    </dependency>
+    <dependency>
+        <groupId>org.apache.tomcat</groupId>
+        <artifactId>tomcat-el-api</artifactId>
+        <version>8.5.35</version>
+        <scope>provided</scope>
+    </dependency>
+    <dependency>
+        <groupId>org.apache.tomcat</groupId>
+        <artifactId>tomcat-jasper-el</artifactId>
+        <version>8.5.35</version>
+        <scope>provided</scope>
+    </dependency>
+```
+
+4. Create a file with name project-springapp-test.xml in src/test/resources folder
+5. Add below snippet to the file
+
+```
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+  <import resource="file:src/main/webapp/WEB-INF/project-springapp.xml"/>
+</beans>
+```
+
+6. Go to folder src/test/java and create a folder structure com/<appName>/test/core 
```

Comment:
not a folder structure, say package.

Reply:
Changed from "Folder structure" to "Package"